File: python/docmost/services/attachment_search_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import SearchDto


class AttachmentSearchApi:
    """Attachment Search API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def search_attachments(self, request: SearchDto) -> ApiResponse[Any]:
        return self._http.post("search-attachments", body=request.model_dump(by_alias=True, exclude_none=True))

    def trigger_attachment_indexing(self) -> ApiResponse[Any]:
        return self._http.post("search-attachments/indexing", body=None)

File: go/docmost/services/attachment_search_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// AttachmentSearchAPI implements Attachment Search API.
type AttachmentSearchAPI struct {
	http *httpclient.HTTPClient
}

func NewAttachmentSearchAPI(http *httpclient.HTTPClient) *AttachmentSearchAPI {
	return &AttachmentSearchAPI{http: http}
}

func (a *AttachmentSearchAPI) SearchAttachments(ctx context.Context, request *models.SearchDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "search-attachments", request)
}

func (a *AttachmentSearchAPI) TriggerAttachmentIndexing(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "search-attachments/indexing", nil)
}

